docs/decisions/0039-template-engine.md

```diff
@@ -0,0 +1,74 @@
+---
+nav_order: 39
+parent: Decision Records
+---
+
+# Template engine
+
+## Context and Problem Statement
+
+We need to choose a template engine for [custom export filters](https://docs.jabref.org/collaborative-work/export/customexports) and [AI features](https://github.com/JabRef/jabref/pull/11884).
+
+A discussion of template engines was also in [one of the JabRef repos](https://github.com/koppor/jabref/issues/392).
+
+A discussion was raised on StackOverflow ["Velocity vs. FreeMarker vs. Thymeleaf"](https://stackoverflow.com/q/1459426/10037342).
+
+## Decision Drivers
+
+* It should support different formats, in case JabRef changes in the future. Also, AI features require just plain text manipulation.
+* It should have all necessary constructs from programming languages: loops, conditions, variables. It should support list or object (map) variables.
+* It should be embeddable and small. AI features require that templates could be generated out of simple `String` type.
+
+## Considered Options
+
+* Apache Velocity
+* Apache FreeMarker
+* Thymeleaf
+
+## Decision Outcome
+
+Chosen option: "{title of option 1}", because
+{justification. e.g., only option, which meets k.o. criterion decision driver | which resolves force {force} | … | comes out best (see below)}.
+
+## Pros and Cons of the Options
+
+### Apache Velocity
+
+Main page: <https://velocity.apache.org/>.
+User guide: <https://velocity.apache.org/engine/devel/user-guide.html>.
+Developer guide: <https://velocity.apache.org/engine/devel/developer-guide.html>.
```

Comment:
Please use Markdown lists, otherwise, it is rendred badly

Use `-` as lsit symbol, then GitHub seperates this list from the other list.

Add following to the end of the file

    <!-- markdownlint-disable-file MD004 -->

---

Do that also for the other options.

Reply:
Please adress this!

Reply:
But everywhere else we use `*` for lists